spk/salt-minion/src/service-setup.sh

```diff
@@ -1,33 +1,33 @@
-PYTHON_DIR="/var/packages/python3/target/bin"
-PATH="${SYNOPKG_PKGDEST}/bin:${SYNOPKG_PKGDEST}/env/bin:${PYTHON_DIR}:${PATH}"
+PYTHON_DIR="/bin"
+PATH="${SYNOPKG_PKGDEST}/bin:${PATH}"
 VIRTUALENV="${PYTHON_DIR}/python3 -m venv"
-PYTHON="${SYNOPKG_PKGDEST=}/env/bin/python"
+PYTHON="/bin/python3"
 LANGUAGE="env LANG=en_US.UTF-8"
-SALT_MINION="${SYNOPKG_PKGDEST}/env/bin/salt-minion"
+SALT_MINION="${SYNOPKG_PKGDEST}/bin/salt-minion"
 PID_FILE="${SYNOPKG_PKGDEST}/var/run/salt-minion.pid"
 
-SERVICE_COMMAND="${SALT_MINION} -c ${SYNOPKG_PKGDEST}/var -d"
+SERVICE_COMMAND="${SALT_MINION} -c ${SYNOPKG_PKGDEST}/etc/salt -d"
 
 service_postinst ()
 {
     # Create a Python virtualenv
-    ${VIRTUALENV} --system-site-packages ${SYNOPKG_PKGDEST}/env
+    ${VIRTUALENV} --system-site-packages ${SYNOPKG_PKGDEST}
 
-    # Install wheels
+    # Install the wheels
     wheelhouse=${SYNOPKG_PKGDEST}/share/wheelhouse
-    ${SYNOPKG_PKGDEST}/env/bin/pip install --no-deps --force-reinstall --no-index --find-links ${wheelhouse} ${wheelhouse}/*.whl
+    ${SYNOPKG_PKGDEST}/bin/pip install --no-deps --no-index -U --force-reinstall -f ${wheelhouse} ${wheelhouse}/*.whl
 
     # Patch rsax931.py file to find libcrypto lib
     # (Rely on patch util bundled with python3's busybox)
-    ${PYTHON_DIR}/bin/patch ${SYNOPKG_PKGDEST}/env/lib/python3.7/site-packages/salt/utils/rsax931.py < ${SYNOPKG_PKGDEST}/share/rsax931.py.patch
+    ${SYNOPKG_PKGDEST}/bin/patch ${SYNOPKG_PKGDEST}/lib/python*/site-packages/salt/utils/rsax931.py -i ${SYNOPKG_PKGDEST}/share/rsax931.py.patch
 
-    # Prepare salt-minion config in /var/salt
-    install -m 755 -d ${SYNOPKG_PKGDEST}/var
-    install -m 755 -d ${SYNOPKG_PKGDEST}/var/minion.d
-    install -m 644 ${SYNOPKG_PKGDEST}/share/minion.conf ${SYNOPKG_PKGDEST}/var
-    echo "pidfile: ${PID_FILE}" > ${SYNOPKG_PKGDEST}/var/minion.d/02_pidfile.conf
+    # Prepare salt-minion config in /var/packages/salt-minion/
+    install -m 755 -d ${SYNOPKG_PKGDEST}/etc/salt/minion.d
```

Comment:
I would like to keep the `var` folder instead of `etc` folder to keep DSM 6 compatibility and the automatic backup/restore on package update.
@publicarray what are your thoughts?

Reply:
What about I do  seperate dsm6 and dsm7 service-setup.sh ?
I prefer to follow synology best practices for dsm7

Reply:
It's Ok to separate it, but the update from DSM 6 to DSM 7 will not be straight forward. 

BTW I will be offline for some weeks (at least i have no access to my dev. environment) but @publicarray and @th0ma7 have good experiance with the spksrc framework.

Reply:
@hgy59 which DSM 6 version are you refering to?

bjarne@dsm6:/var/packages/open-vm-tools$ cat /etc.defaults/VERSION
majorversion="6"
minorversion="2"
productversion="6.2.3"
buildphase="GM"
buildnumber="25426"
smallfixnumber="0"
builddate="2020/05/12"
buildtime="06:14:27"
bjarne@dsm6:/var/packages/open-vm-tools$ ls -la
total 24
drwxr-xr-x 4 root root 4096 Sep  4 13:20 .
drwxr-xr-x 7 root root 4096 Sep  4 13:27 ..
drwxr-xr-x 2 root root 4096 Sep  4 13:23 conf
-rw-r--r-- 1 root root    0 Sep  4 13:20 enabled
lrwxrwxrwx 1 root root   36 Sep  4 13:20 etc -> /usr/syno/etc/packages/open-vm-tools
-rw-r--r-- 1 root root 6709 Sep  4 13:20 INFO
drwxr-xr-x 2 root root 4096 Nov 28  2019 scripts
lrwxrwxrwx 1 root root   32 Sep  4 13:20 target -> /volume1/@appstore/open-vm-tools
bjarne@dsm6:/var/packages/open-vm-tools$

DSM 6.2.3 follow the same layout as >= DSM 7.

Reply:
I'm ok with us starting to use `etc` for small config files. It's just more paths to remember to migrate between version upgrades, for end users to know and for us to document.  The `var` folder already has some scripts that make upgrades more convenient.